apis/v1alpha1/httproute_types.go

```diff
@@ -77,6 +98,23 @@ type HTTPRouteHost struct {
 	ExtensionRef *RouteHostExtensionObjectReference `json:"extensionRef"`
 }
 
+// RouteTLSConfig describes a TLS configuration defined at the Route level.
+type RouteTLSConfig struct {
+	// CertificateRef refers to a Kubernetes object that
+	// contains a TLS certificate and private key.
+	// This certificate MUST be used for TLS handshakes for the domain
+	// this RouteTLSConfig is associated with.
+	// If an entry in this list omits or specifies the empty
+	// string for both the group and the resource, the resource defaults to "secrets".
+	// An implementation may support other resources (for example, resource
+	// "mycertificates" in group "networking.acme.io").
+	// Support: Core (Kubernetes Secrets)
+	// Support: Implementation-specific (Other resource types)
+	//
+	// +required
+	CertificateRef CertificateObjectReference `json:"certificateRef,omitempty"`
```

Comment:
@hbagdi `omitempty` should be removed when the field is required and defaulting is not being used.